### PR TITLE
Allow enabling HTTP Application Routing Addon

### PR DESCRIPTION
**Pull request by hairmare**
Makes the HTTP Application Routing Addon configurable through the `enable_http_application_routing` variable.

### CODE REVIEW

**Comment by hairmare** [comment]
I just rebased this to make it merge-able again.

**Comment by n01d** [comment]
Hi @hairmare , I am very sorry it took me so long to come back to this. Could you please update to terraform 0.12 syntax / update with the lastest changes so I can test and merge your addition to the module?

File: modules/kubernetes-cluster/outputs.tf [hunk 1]
```diff
@@ -37,3 +37,7 @@ output "node_resource_group" {
 output "location" {
   value = "${azurerm_kubernetes_cluster.main.location}"
 }
+
+output "http_application_routing_zone_name" {
+  value = "${azurerm_kubernetes_cluster.main.addon_profile.0.http_application_routing.0.http_application_routing_zone_name}"
```
Comment (sblack4) [inline modules/kubernetes-cluster/outputs.tf]: Does this throw an error if there is no routing zone? We might want to add the same conditional as used in the creation (I haven't tested this though)
```suggestion
  value = var.enable_http_application_routing ? azurerm_kubernetes_cluster.main.addon_profile.0.http_application_routing.0.http_application_routing_zone_name : ""
```

**Comment by hairmare** [comment]
I did a quick rebase to get rid of the merge conflicts but didn't look into the review comment from @sblack4.

**Comment by yupwei68** [comment]
Hi @hairmare  , thanks for opening this PR 🚀 Would you mind adding one example in test/fixture/main.tf and one in README.md?

**Comment by yupwei68** [comment]
Thanks for opening this PR. This has been released in 4.1.0

**Pull request closed without merging**